### DISCUSSION TITLE
size of library

### DISCUSSION THREAD

**Discussion by Jaredkzk** [v923z/micropython-ulab#587] [Q&A]
How big is this library. I'm trying to install it onto a Raspberry Pi Pico with RO2040 and there is an management error every single time. I'm using Thonny to install the library. It will make folders on the Pico but none of the actual code or files seems to being transferred. I can import numpy as np but cant call any functions. Here is a file with the shell output when i try to install the library. 
[numpyerror2-5-23.txt](https://github.com/v923z/micropython-ulab/files/10612002/numpyerror2-5-23.txt)

**Comment by jimmo** [comment]
@Jaredkzk 
ulab needs to be compiled into the MicroPython firmware. I'm not sure exactly the details for whether there are pre-built images for the Pico, but see https://github.com/v923z/micropython-ulab for more information.

What you have done is attempted to install the "real" `numpy` library. Thonny presents every possible CPython package from PyPI, although the vast majority are not at all likely to work in MicroPython.

**Comment by v923z** [comment]
@Jaredkzk I don't know which exact hardware platform you have, but you can find compiled firmware here: https://github.com/v923z/micropython-builder/. If your platform is missing, I would be more than happy to add that.

**Reply by Jaredkzk** [reply]
I'm using Windows and Thonny to try and code a Raspberry Pi Pico
